src/objective_coefficients.jl

```diff
@@ -0,0 +1,105 @@
+"""
+    set_dualmodel_sense!(dual_model::MOI.ModelLike, model::MOI.ModelLike)
+
+Set the dual model objective sense
+"""
+function set_dual_model_sense(dual_model::MOI.ModelLike, primal_model::MOI.ModelLike) where T
+    # Get model sense
+    primal_sense = MOI.get(primal_model, MOI.ObjectiveSense())
+    if primal_sense == MOI.FEASIBILITY_SENSE
+        error(primal_sense, " is not supported") # Feasibility should be supported?
+    end
+    # Set dual model sense
+    dual_sense = (primal_sense == MOI.MIN_SENSE) ? MOI.MAX_SENSE : MOI.MIN_SENSE
+    MOI.set(dual_model, MOI.ObjectiveSense(), dual_sense)
+    return 
+end
+
+# Primals
+"""
+    PrimalObjectiveCoefficients{T}
+
+Primal objective coefficients defined as ``a_0^Tx + b_0`` as in
+http://www.juliaopt.org/MathOptInterface.jl/stable/apimanual/#Advanced-1
+"""
+struct PrimalObjective{T}
+    saf::SAF{T}
+end
+
+"""
+    get_primal_obj_coeffs(model::MOI.ModelLike)
+
+Get the coefficients from the primal objective function and
+return a `PrimalObjectiveCoefficients{T}`
+"""
+function get_primal_objective(primal_model::MOI.ModelLike) where T
+    return _get_primal_objective(primal_model.objective)
```

Comment:
`.objective` won't work for a `ModelLike`. You need to do `MOI.get(model, MOI.get(model, ObjectiveFunctionType()))`